core/src/main/java/org/apache/iceberg/DataTableScan.java

```diff
@@ -61,6 +61,19 @@ public TableScan appendsAfter(long fromSnapshotId) {
     return appendsBetween(fromSnapshotId, currentSnapshot.snapshotId());
   }
 
+  @Override
+  public TableScan useBranch(String name) {
```

Comment:
@amogh-jahagirdar do theseneed to be implemented in BaseTableScan? Just following acc to `asOfTime`

Reply:
Yes good point. we should have a default implementation (well it's an abstract class, so just an implementation) which throws Unsupported. Right now DataTableScan would be the only table scan which supports branching and tagging.

Reply:
I finally got a chance to come back to this PR,  I actually decided on the opposite approach. There's fewer table scans where we want to fail, so I think it makes sense to have the implementation in the parent class and then in the child classes which don't support (like all metadata table scan, or incremental scan) we throw an unsupported exception.